packages/@adobe/spectrum-css-temp/components/searchwithin/index.css

```diff
@@ -11,49 +11,59 @@ governing permissions and limitations under the License.
 */
 
 @import '../commons/index.css';
-
-:root {
+.spectrum-SearchWithin {
   --spectrum-searchwithin-width: 250px;
 
   /* Force override */
   --spectrum-searchwithin-border-radius: 0;
-  --spectrum-searchwithin-dropdown-min-width: 0;
+  --spectrum-searchwithin-picker-min-width: 0;
 }
 
 .spectrum-SearchWithin {
-  width: var(--spectrum-searchwithin-width);
+  inline-size: var(--spectrum-searchwithin-width);
   display: inline-flex;
   position: relative;
+}
 
-  .spectrum-Dropdown {
-    width: auto;
-    min-width: 0;
-  }
-
-  .spectrum-Dropdown-trigger {
-    border-top-right-radius: var(--spectrum-searchwithin-border-radius);
-    border-bottom-right-radius: var(--spectrum-searchwithin-border-radius);
-  }
+.spectrum-SearchWithin-picker {
+  inline-size: auto;
+  min-inline-size: 0;
+  flex-shrink: 0;
 
-  .spectrum-Dropdown-label {
-    /* Override dropdown's min-width and be tiny */
-    min-width: var(--spectrum-searchwithin-dropdown-min-width);
+  button {
+    width: unset;
+    contain: unset;
+    justify-content: unset;
+    border-end-start-radius: var(--spectrum-searchwithin-border-radius);
+    border-start-start-radius: var(--spectrum-searchwithin-border-radius);
   }
+}
 
-  .spectrum-Textfield {
-    flex: 1;
-    margin-left: calc(var(--spectrum-textfield-border-size) * -1); /* hides left border */
-    border-top-left-radius: var(--spectrum-searchwithin-border-radius);
-    border-bottom-left-radius: var(--spectrum-searchwithin-border-radius);
+.spectrum-SearchWithin-input {
+  flex: 1;
+  box-sizing: border-box;
+  min-inline-size: inherit;
+  margin-inline-end: calc(
+    -1 * var(--spectrum-textfield-border-size)
+  ); /* hides left border */
 
-    &:hover, &:focus {
-      position: relative; /* shows left border */
-    }
+  input {
+    border-end-end-radius: var(--spectrum-searchwithin-border-radius);
+    border-start-end-radius: var(--spectrum-searchwithin-border-radius);
   }
 
-  .spectrum-ClearButton {
-    position: absolute;
-    top: 0;
-    right: 0;
+  &:hover,
+  &:focus {
+    position: relative; /* shows left border */
   }
 }
+
+.spectrum-SearchWithin-clearButton {
+  position: absolute;
+  inset-block-start: 0;
+  inset-inline-end: 0;
+}
+
+.spectrum-SearchWithin > .spectrum-Textfield {
```

Comment:
This selector won't work, just a heads up, we use CSS Modules https://github.com/css-modules/css-modules
every classname defined in this file will get a hash appended to it that is different from classes in another file
so textfield.css might have `spectrum-Textfield` -> `spectrum-Textfield_582e1` while searchwithin.css would have `spectrum-Textfield` -> `spectrum-Textfield_ca532`
I think this is the reason you're using selectors like `input` and `button` as well?